File: solvers/_common/disp_stats.m
% display statistics
function disp_stats(stats,hist)
    if ~exist('hist','var'); hist = 0; end
    g = '30'; % gap
    fprintf(['%-', g, 's %-', g, 's\n'],'Problem:',stats.problem);
    fprintf(['%-', g, 's %-', g, '.1f\n'],'Average runtime (ms):',10^3*mean(stats.tm));
    fprintf(['%-', g, 's %-', g, '.2f\n'],'Mean num. error:',mean(stats.err));
    fprintf(['%-', g, 's %-', g, '.2f\n'],'Median num. error:',median(stats.err));
    fprintf(['%-', g, 's %-', g, 'd\n'],'Min. number of roots:',min(stats.k));
    fprintf(['%-', g, 's %-', g, 'd\n'],'Max. number of real roots:',max(stats.kr));
    fprintf(['%-', g, 's %-', g, '.1f\n'],'Ave. number of real roots:',mean(stats.kr));
    fprintf(['%-', g, 's %-', g, '.1f\n'],'Fails (%):',(1-length(stats.k)/stats.N)*100);
    if hist==1; plot_histo_num(stats); end
end

File: solvers.julia/_common/disp_stats.jl
# display solver statistics
function disp_stats(stats)

    @printf "\n%-30s %-30s\n" "Problem:" stats.name
    @printf "%-30s %-30.2f\n" "Ave. runtime (ms):" 10^3*Statistics.mean(stats.tm)
    @printf "%-30s %-30.2f\n" "Max. num. error:" maximum(stats.err)
    @printf "%-30s %-30.2f\n" "Ave. num. error:" Statistics.mean(stats.err)
    @printf "%-30s %-30.2f\n" "Med. num. error:" Statistics.median(stats.err)
    @printf "%-30s %-30.1f\n" "Fails (%):" Statistics.mean(stats.flr)*100
    @printf "%-30s %-30.2f\n" "Ave. sharpness:" Statistics.mean(stats.shrp)
    @printf "%-30s %-30.2f\n" "Med. sharpness:" Statistics.median(stats.shrp)
    @printf "%-30s %-30d\n" "Min. number of roots:" minimum(stats.k)
    @printf "%-30s %-30d\n" "Max. number of real roots:" maximum(stats.kr)
    @printf "%-30s %-30.1f\n" "Ave. number of real roots:" Statistics.mean(stats.kr)

end